onnxruntime/core/providers/cpu/controlflow/utils.h

```diff
@@ -7,12 +7,14 @@
 #include <string>
 #include <vector>
 
-#include "core/common/common.h"
 #include "core/framework/feeds_fetches_manager.h"
+#ifdef SHARED_PROVIDER
```

Comment:
It's not clear why I would check for this macro in the implementation of a CPU op. Is this documented somewhere?

Reply:
Many of the cuda ops will share functions with the cpu ops, so they '#include' the cpu headers to access these methods, which will fail on the internal headers that don't work from a shared provider.

Typically it's either a base class, a structure, or parameter validation methods the cuda providers share with the cpu one. These functions are routed across the bridge.

Reply:
Sure. Would be good to comment this somewhere so that people are not left guessing.